config/routes.rb

```diff
@@ -49,6 +49,7 @@
   get '/users/sign_in', to: redirect('/sign-in')
   get '/users/sign_up', to: redirect('/sign-up')
 
+  post '/notifications/mark_read(/:notification_id)' => 'notifications#mark_read'
   resources :notifications, only: [:index, :show]
```

Comment:
Again, if we use REST for this instead, it'll need the addition of `:update` to the `:notifications` resources and removal of the `/mark_read/` endpoint.

But if we do use RPC, it should also be `/notifications/:id/mark_read`, since `mark_read` is a method of the notification — not the other way around.